MapboxMobileEvents/MMEEvent.m

```diff
@@ -296,41 +317,41 @@ - (instancetype)init {
 }
 
 - (instancetype)initWithAttributes:(NSDictionary *)eventAttributes error:(NSError **)error {
-    if (eventAttributes == MMEEvent.nilAttributes) { // special case for initFromCoder
-        self = [super init];
-    }
-    else if ([NSJSONSerialization isValidJSONObject:eventAttributes]) {
-        if (![eventAttributes.allKeys containsObject:MMEEventKeyEvent]) { // is required
-            *error = [NSError errorWithDomain:MMEErrorDomain code:MMEErrorEventInit userInfo:@{
-                MMEErrorDescriptionKey: @"eventAttributes does not contain MMEEventKeyEvent",
-                MMEErrorEventAttributesKey: eventAttributes
-            }];
-            self = nil;
+    @try {
+        if (eventAttributes == MMEEvent.nilAttributes) { // special case for initFromCoder
```

Comment:
can `eventAttributes` be `nil`?

Reply:
The header has `NS_ASSUME_NONNULL_BEGIN` enabled, so it'll generate a warning at compile time if a real nil is provided. `MMEEvent.nilAttributes` is a special object we use to allow for constructing an event which we can then fill in with initFromCoder.